joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/AstCreationPass.scala

```diff
@@ -36,9 +36,11 @@ class AstCreationPass(codeDir: String, filenames: List[String], cpg: Cpg, keyPoo
     val parser = new JavaParser(parserConfig)
     val parseResult = parser.parse(new java.io.File(filename))
 
+    val typeInfoProvider = TypeInfoProvider(global)
```

Comment:
soft suggestion, I recently renamed `kotlin2cpg`s _TypeInfoProvider_ to _NameGenerator_ because I think the name fits better https://github.com/joernio/joern/commit/b30c221fd0e8fbf1ba21637684e22df5ce67453e

Reply:
The PR where you renamed it was actually what inspired this PR. I was undecided on which name to choose for this one, but figured _TypeInfoProvider_ fit better in this situation (since they are slightly different and this one doesn't exactly _generate_ names per se). I'm not convinced that _TypeInfoProvider_ is the best name in this case either, with something like _TypeResolver_ being closer to accurate, but it doesn't seem overly confusing, at least.